packages/amplify-codegen/src/utils/GraphQLStatementsFormatter.js

```diff
@@ -36,27 +51,42 @@ class GraphQLStatementsFormatter {
   }
 
   formatGraphQL(statements) {
-    const headerBuffer = this.headerComments.map( comment => `# ${comment}`).join(LINE_DELIMITOR);
+    const headerBuffer = this.headerComments.map(comment => `# ${comment}`).join(LINE_DELIMITOR);
     const statementsBuffer = statements ? [...statements.values()].join(LINE_DELIMITOR) : '';
     const formattedOutput = [headerBuffer, LINE_DELIMITOR, statementsBuffer].join(LINE_DELIMITOR);
     return formattedOutput;
   }
 
   formatJS(statements) {
     const lintOverridesBuffer = this.lintOverrides.join(LINE_DELIMITOR);
-    const headerBuffer = this.headerComments.map( comment => `// ${comment}`).join(LINE_DELIMITOR);
+    const headerBuffer = this.headerComments.map(comment => `// ${comment}`).join(LINE_DELIMITOR);
     const formattedStatements = [];
     if (statements) {
       for (const [key, value] of statements) {
-        formattedStatements.push(
-          `export const ${key} = /* GraphQL */ \`${value}\``
-        );
+        const typeTag = this.buildTypeTag(key);
+        formattedStatements.push(`export const ${key} = /* GraphQL */ \`${value}\`${typeTag}`);
       }
     }
-    const formattedOutput = [lintOverridesBuffer, headerBuffer, LINE_DELIMITOR, ...formattedStatements].join(LINE_DELIMITOR);
+    const formattedOutput = [lintOverridesBuffer, headerBuffer, LINE_DELIMITOR, this.typeDefs, LINE_DELIMITOR, ...formattedStatements].join(
+      LINE_DELIMITOR,
+    );
     return formattedOutput;
   }
 
+  buildTypeTag(name) {
+    if (!this.opTypeName) return '';
+    if (this.language !== 'typescript') return '';
+
+    const titleCasedName = `${name[0].toUpperCase()}${name.slice(1)}`;
+    const variablesTypeName = `APITypes.${titleCasedName}${this.opTypeName}Variables`;
+    const resultTypeName = `APITypes.${titleCasedName}${this.opTypeName}`;
```

Comment:
I'll have a rev by standup tomorrow that uses `interfaceNameFromOperation` + a new `interfaceVariablesNameFromOperation` instead of re-deriving those names here. If I'm reading things correctly, that should guard against future mismatches due to drift.

Reply:
Added as another PR to compare/contrast: https://github.com/aws-amplify/amplify-codegen/pull/623/files